addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureFileInclusionScanRuleUnitTest.java

```diff
@@ -85,4 +101,141 @@ void shouldReturnExpectedExampleAlert() {
         assertThat(alert.getRisk(), is(equalTo(Alert.RISK_HIGH)));
         assertThat(alert.getConfidence(), is(equalTo(Alert.CONFIDENCE_MEDIUM)));
     }
+
+    @Test
+    void shouldSkipUrlParams() {
+        // Given
+        HttpMessage msg = createMessage("/test/");
+        NameValuePair nvp = new NameValuePair(NameValuePair.TYPE_URL_PATH, "inc", "", 0);
+        // When
+        rule.scan(msg, nvp);
+        // Then
+        assertThat(countMessagesSent, is(equalTo(0)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"", "/"})
+    void shouldSkipIfNoOrEmptyPath(String path) {
+        // Given
+        HttpMessage msg = createMessage(path);
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(countMessagesSent, is(equalTo(0)));
+    }
+
+    @Test
+    void shouldNotContinueIfRandomBodyIsEmpty() throws IOException {
+        // Given
+        String path = "/shouldNotContinueIfRandomBodyIsEmpty";
+        nano.addHandler(new FiHandler(path, Response.Status.OK, ""));
+        HttpMessage msg = this.getHttpMessage(path + "?inc=bar");
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent.size(), is(equalTo(2)));
```

Comment:
Not sure asserting exact number of requests is good here, if the rule is improved and it now sends more/less we have to go through all of them again, if you want to check that the rule is actually sending something check it's not empty.